openfisca_web_site/templates/france/fr/tree/demo.mako

```diff
@@ -0,0 +1,245 @@
+    <script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>
+    <script type="text/javascript">
+
+      //Variables globales
+      var salary1 = 25000;
+      var salary2 = 15000;
+
+      window.addEventListener("load", callOnLoad, false);
+
+      var updateSalaries = _.debounce(function(salary, montant) {
+          if (salary == "salary1") {
+            salary1 = Number(montant);
+          } else if (salary == "salary2") {
+            salary2 = Number(montant);
+          }
+          callOpenFiscaAPI();
+      }, 500);
+
+
+      function formatJSONOfSituation(){
+        return JSON.stringify({
+          "individus": {
+            "Bill": {
+              "salaire_de_base": {
+                "2017": salary1
+              },
+              "revenus_du_travail": {
+                "2017": null
+              }
+            },
+            "Bob": {
+              "salaire_de_base": {
+                "2017": salary2
+              },
+              "revenus_du_travail": {
+                "2017": null
+              }
+            }
+          },
+          "familles": {
+            "famille_1": {
+              "parents": [
+                "Bill",
+                "Bob"
+              ],
+              "prestations_sociales": {
+                "2017": null
+              }
+            }
+          },
+          "foyers_fiscaux": {
+            "foyer_fiscal_1": {
+              "declarants": [
+                "Bill",
+                "Bob"
+              ]
+            }
+          },
+          "menages": {
+            "menage_1": {
+              "conjoint": [
+                "Bob"
+              ],
+              "personne_de_reference": [
+                "Bill"
+              ],
+              "impots_directs": {
+                "2017": null
+              },
+              "revenu_disponible": {
+                "2017": null
+              }
+            }
+          }
+        });
+      }
+
+      function callOnLoad(){
+        var resultsOnLoad = {};
+        resultsOnLoad.salaires_de_base = salary1 + salary2;
+        resultsOnLoad.revenu_disponible = 29191;
+        resultsOnLoad.impots_directs = 3064;
+        resultsOnLoad.prestations_sociales = 0;
+        resultsOnLoad.revenus_du_travail = resultsOnLoad.revenu_disponible + resultsOnLoad.impots_directs;
+        resultsOnLoad.cotisations_salariales = resultsOnLoad.salaires_de_base - resultsOnLoad.revenus_du_travail;
+
+        displayData(resultsOnLoad);
+      }
+
+      function callOpenFiscaAPI(){
+        var xmlhttp = new XMLHttpRequest();
+        var url = "http://localhost:5000/calculate";
+        var json = formatJSONOfSituation();
+        xmlhttp.open("POST", url, true);
+
+        xmlhttp.onreadystatechange=function() {
+          if (xmlhttp.readyState==4 && xmlhttp.status==200) {
+            var responseFromAPI = JSON.parse(xmlhttp.responseText);
+            transformData(responseFromAPI);
```

Comment:
This architecture is not great : every function must know what is the next function to call... and it's not natural for a function named `callOpenFiscaAPI ` to end up calling a function that process results, and another that display them.

The JS way of handling this is callbacks:

```JS
function callOpenFiscaAPI(callback) {
    ...
    var responseFromAPI = JSON.parse(xmlhttp.responseText);
    callback(responseFromAPI)
}
```

and in `updateSalaries `:
```JS
callOpenFiscaAPI(function(response) {
    results = transformData(response)
    displayData(results)
});
```